Preserve spaces in Count1 (tools/filters/uniq.xml)

peterjc opened the pull request:
See https://trello.com/c/Ectva3pr/2510-count1-tool-removes-spaces-tools-filters-uniq-xml

Includes test case showing the change in behaviour from this:

```
$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2.tabular -c 2 -d T
Count of unique values in c2,
1   Ancylostomaceylanicum
4   Ascarissuum
2   Brugiamalayi
1   Caenorhabditiselegans
1   Caenorhabditisremanei
3   Loaloa
1   Meloidogyneincognita
4   NoBLASThit
1   Prunusmume
1   Strongyloidesratti
1   Wuchereriabancrofti
```

To this space-preserving output:

```
$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2.tabular -c 2 -d T
Count of unique values in c2
1   Ancylostoma ceylanicum
4   Ascaris suum
2   Brugia malayi
1   Caenorhabditis elegans
1   Caenorhabditis remanei
3   Loa loa
1   Meloidogyne incognita
4   No BLAST hit
1   Prunus mume
1   Strongyloides ratti
1   Wuchereria bancrofti
```

Also includes a multi-column test to confirm that still works nicely.

Also makes the tool version number explicit (would have defaulted to `1.0.0` so now using `1.0.1` as per IUC mailing list discussion).

Also removes the trailing comma in the tool's stdout.

peterjc committed:
Do not remove spaces in Count1 (tools/filters/uniq.xml), with tests

Also sets the tool version for the first time, using 1.0.1.

$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2.tabular -c 2 -d T
Count of unique values in c2,

$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2c3.tabular -c 2,3 -d T
Count of unique values in c2, c3,

peterjc committed:
Remove trailing comma space in uniq.py stdout

$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2.tabular -c 2 -d T
Count of unique values in c2,

$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2c3.tabular -c 2,3 -d T
Count of unique values in c2, c3,

becomes:

$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2.tabular -c 2 -d T
Count of unique values in c2

$ python tools/filters/uniq.py -i test-data/species_assignment.tabular -o test-data/species_assignment_c2c3.tabular -c 2,3 -d T
Count of unique values in c2, c3

jmchilton commented:
:+1:

davebx commented:
:+1: from me, as well.

dannon commented:
This is an automated message. Thanks for your contribution, a [Trello card](https://trello.com/c/oSmNO45W) to track this issue has been created. [Apply this patch](https://github.com/galaxyproject/galaxy/pull/13.patch) for testing.

peterjc commented:
Does it need an example in the help? My motivating use case (reflected in the test example) is to prepare a tally table for use with the Pie Chart visualisation tool (where the removal of the spaces is quite annoying).

peterjc committed:
Fix trailing space removal

I had a test case which showed the problem, I just over looked it.

The input file species_assignment.tabular now includes more trailing
space examples columns 2, 3 and 4 to make this visually clearer at
first glance (previously the single trailing space example was
too easily overlooked).

peterjc commented:
I'm willing to add leading space removal (and even collapsing double spaces to single spaces) as a nod towards the original space-removing behaviour. My thinking in adding trailing space removal was practicality as they are effectively invisible in most editors.

I'm also willing to remove all the space cleaning - it would make the script a lot simpler without the sed calls.

jmchilton commented:
Tweaked and merged directly, see conversation on #16.
